### ISSUE TITLE
letsencrypt dns-directadmin not working anymore

### ISSUE THREAD

**Issue by DirkKalkman** [home-assistant/addons#3784]
### Describe the issue you are experiencing

certbot-dns-directadmin packed is version 1.04
This does not work with directadmin anymore and produces an error:
Encountered exception during recovery: ValueError: too many values to unpack (expected 2)

This is fixed in version 1.06

### What type of installation are you running?

Home Assistant Supervised

### Which operating system are you running on?

Debian

### Which add-on are you reporting an issue with?

Let's Encrypt

### What is the version of the add-on?

5.2.1

### Steps to reproduce the issue

1. Request certificate with dns-directadmin

## System Information

version | core-2024.10.0
-- | --
installation_type | Home Assistant Supervised
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.4
os_name | Linux
os_version | 6.6.47+rpt-rpi-2712
arch | aarch64
timezone | Europe/Amsterdam
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
HACS Data | ok
GitHub API Calls Remaining | 5000
Installed Version | 2.0.1
Stage | running
Available Repositories | 1428
Downloaded Repositories | 1

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Debian GNU/Linux 12 (bookworm)
-- | --
update_channel | stable
supervisor_version | supervisor-2024.09.1
agent_version | 1.6.0
docker_version | 27.3.1
disk_total | 200.0 GB
disk_used | 10.4 GB
healthy | true
supported | true
host_connectivity | true
supervisor_connectivity | true
ntp_synchronized | true
virtualization | 
supervisor_api | ok
version_api | ok
installed_addons | NGINX Home Assistant SSL proxy (3.11.0), Vaultwarden (Bitwarden) (0.23.0), File editor (5.8.0), Get HACS (1.3.1), Syncthing (1.19.0), Let's Encrypt (5.2.1)

</details>

<details><summary>Dashboards</summary>

dashboards | 2
-- | --
resources | 0
views | 0
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | September 26, 2024 at 4:54 PM
-- | --
current_recorder_run | October 3, 2024 at 9:31 AM
estimated_db_size | 29.37 MiB
database_engine | sqlite
database_version | 3.45.3

</details>

### Anything in the add-on logs that might be useful for us?

```txt
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun lets-encrypt (no readiness notification)
s6-rc: info: service legacy-services successfully started
[15:12:16] INFO: Selected DNS Provider: dns-directadmin
[15:12:16] INFO: Use propagation seconds: 300
[15:12:16] INFO: Detecting existing certificate type for jarvis.kalkman.eu
Saving debug log to /var/log/letsencrypt/letsencrypt.log
[15:12:17] INFO: No certificate found - using 'ecdsa' key type.
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Requesting a certificate for jarvis.kalkman.eu
Encountered exception during recovery: ValueError: too many values to unpack (expected 2)
An unexpected error occurred:
ValueError: too many values to unpack (expected 2)
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Comment by mantaalex** [comment]
I have the same problem i hope someone can fix it

**Comment by mantaalex** [comment]
anyone an idea to fix this manually?

**Comment by Rolfieo** [comment]
Looks like i'm not the only one. I have indeed exactly the same issue. 

```
Encountered exception during recovery: ValueError: too many values to unpack (expected 2)
An unexpected error occurred:
ValueError: too many values to unpack (expected 2)
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile 
```

**Comment by sloetr** [comment]
I encountered the same issue. 

For me, the temporary fix was to downgrade to version 5.1.0. This can be done by taking a backup of the add-on, replace the version number in the backup files with 5.1.0 (3 times in my situation) and then restore that backup.

**Comment by MathieuIci** [comment]
If your certificate is about to expire and need a temporary fix, the [above](https://github.com/home-assistant/addons/issues/3784#issuecomment-2449524689) suggestion worked for me.  I downgraded the Let's Encrypt add-on to version 5.1.0 by restoring an old backup and now all is well:

![image](https://github.com/user-attachments/assets/a2b4f708-b22e-4fab-9524-726deee442b0)

**Comment by peterwesterlund** [comment]
> I encountered the same issue.
> 
> For me, the temporary fix was to downgrade to version 5.1.0. This can be done by taking a backup of the add-on, replace the version number in the backup files with 5.1.0 (3 times in my situation) and then restore that backup.

Would you mind sharing your tar file? I have tried myself but I can't get it tor work. I  suspect something gets wrong when I compress it back and forth.

**Comment by dakiers** [comment]
I hoped this was fixed in  5.2.5 but no dice (for me)

**Comment by MathieuIci** [comment]
Issue is still present on the latest version 5.2.10.

**Comment by GoetzGoerisch** [comment]
Could you please verify if this is fixed by using this version: https://github.com/GoetzGoerisch/dev-addon

Steps:
- Install as separate addon repo (Addon-Store -> Repository)
- Copy config from official plugin
- run the plugin

**Comment by dakiers** [comment]
> Could you please verify if this is fixed by using this version: https://github.com/GoetzGoerisch/dev-addon
> 
> Steps:
> 
> * Install as separate addon repo (Addon-Store -> Repository)
> * Copy config from official plugin
> * run the plugin

Sadly for me still:

>Encountered exception during recovery: urllib.error.HTTPError: HTTP Error 403: Forbidden
>An unexpected error occurred:
>urllib.error.HTTPError: HTTP Error 403: Forbidden

**Comment by GoetzGoerisch** [comment]
@dakiers but HTTP 403 is different from above and points to an authentication issue.
Did you ensure the API key has the correct authentication scope configured?

**Comment by dakiers** [comment]
> [@dakiers](https://github.com/dakiers) but HTTP 403 is different from above and points to an authentication issue. Did you ensure the API key has the correct authentication scope configured?

Yeah I took notice that there's a different Exception for me, though the problems started around the same time as this issue got reported. So I assumed it had similar cause, but I can't rule it it's something else.
Sadly I haven't been able to get a more detailed log.

I can see in direct-admin that
> 2025:02:10-11:30:26:ip={redacted}: /CMD_API_SHOW_DOMAINS: Valid Login
So it happens after that. 

Not sure what you mean with scope. The login key has the 3 commands set (CMD_API_SHOW_DOMAINS, CMD_API_LOGIN_TEST & CMD_API_DNS_CONTROL).

Anyway curious what other people experience with this build.

**Comment by GoetzGoerisch** [comment]
According to [documentation](https://certbot-dns-directadmin.readthedocs.io/) the login key needs these 4 permissions-

```
CMD_API_LOGIN_TEST
CMD_API_DNS_CONTROL
CMD_API_SHOW_DOMAINS
CMD_API_DOMAIN_POINTER
```

Please verify, as 403 is an authentication issue.

**Comment by dakiers** [comment]
> According to [documentation](https://certbot-dns-directadmin.readthedocs.io/) the login key needs these 4 permissions-
> 
> ```
> CMD_API_LOGIN_TEST
> CMD_API_DNS_CONTROL
> CMD_API_SHOW_DOMAINS
> CMD_API_DOMAIN_POINTER
> ```
> 
> Please verify, as 403 is an authentication issue.

Lovely thank you. Works like a charm now for me.

I guess ''CMD_API_DOMAIN_POINTER'' has been more recently since it worked fine without it before and it's not mentioned in the [example documentation](https://github.com/home-assistant/addons/blob/master/letsencrypt/DOCS.md#example-configurations) of the addon.
